src/Microsoft.ML.Featurizers/Common.cs

```diff
@@ -219,5 +220,31 @@ internal static TypeId GetNativeTypeIdFromType(this Type type)
 
             throw new InvalidOperationException($"Unsupported type {type}");
         }
+
+        // The Native Featurizers do not currently support CentOS7, this method checks the OS and returns true if it is CentOS7.
+        internal static bool OsIsCentOS7()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                using (Process process = new Process())
```

Comment:
A more performant way of doing this check would be to read from the file directly. Starting a new process just to get this info is too heavy-weight IMO.

See [https://github.com/dotnet/runtime/blob/6f445d6dc237f59d730e0df47f8630b18887d776/src/installer/managed/Microsoft.DotNet.PlatformAbstractions/Native/PlatformApis.cs#L140-L142](https://github.com/dotnet/runtime/blob/6f445d6dc237f59d730e0df47f8630b18887d776/src/installer/managed/Microsoft.DotNet.PlatformAbstractions/Native/PlatformApis.cs#L140-L142)